Source/WebInspectorUI/UserInterface/Views/ObjectTreeView.css

```diff
@@ -25,7 +25,7 @@
 
 .object-tree {
     position: relative;
-    display: inline-block;
+    display: inline;
```

Comment:
This will apply to every object tree throughout Web Inspector, including the ones used in the Sources tab for the Scope sidebar as well as in the popovers when hovering over a value paused in the debugger. We should confirm those still behave and size as expected with this adjustment.

If not, it seems like it would still be valid to set `display: inline;` for `.console-top-level-message .object-tree`.

Reply:
Good idea. I went ahead and tested the object tree view in the following places:

(All screenshots are "after versus before")

<img width="1873" alt="Screenshot 2024-05-31 at 11 57 11" src="https://github.com/WebKit/WebKit/assets/44916353/d78a0a21-dc23-4aa3-a95f-306e06620a24">
<img width="1758" alt="Screenshot 2024-05-31 at 11 15 30" src="https://github.com/WebKit/WebKit/assets/44916353/61ad6192-7925-4eea-be51-77a09d238590">
<img width="1926" alt="Screenshot 2024-05-31 at 11 46 40" src="https://github.com/WebKit/WebKit/assets/44916353/372bd76c-1d2b-4407-a556-acab9b52abe3">
<img width="1700" alt="Screenshot 2024-05-31 at 11 16 43" src="https://github.com/WebKit/WebKit/assets/44916353/ee5fb1f7-e24b-4b59-bed8-121d3bedfa41">

As seen from the screenshots, this patch also appropriately addresses the same issue with object trees displayed elsewhere in the inspector.

However, there are two notable side effects brought by this patch:

1. Some cut-off lines now show ellipses at the end. I think this is fine because ellipses should generally be expected on lines that don't fit and are unscrollable. Though as it stands there are still lines without ellipses which is rather inconsistent, so this side effect is barely an improvement.
2. In the last two screenshots, the original behavior was that the object trees were scrollable to the right, and this patch made them not scrollable anymore. **I would consider this a regression,** so I think what we can do is let the parents of the object tree decide whether it should be scrollable to the right, by optionally being `display: flex` to allow scrolling.

Reply:
in the case that something isn't horizontally scrollable, i agree that showing "…" is probably better than just letting the content keep going, but in all of those screenshots it looks like the content is cut off _way_ before the end of the container, which is definitely not ideal

unless that's fixed by your most recent changes?

Reply:
Good catch! I added a patch for that so that the cut-off point for any nested text is always at the same verticle boundary, basically by taking out all the `padding-right`s those tree elements used to have.

Reply:
btw i have a more general question

when/why did all this break?  i know this UI was fine for **_years_**, so it's extremely odd for it to suddenly have issues.  makes me wonder if Web Inspector was doing something hacky or if WebKit has changed such that suddenly there's web compatibility issues :/

Reply:
Right, I agree that it might be worthwhile to pinpoint what happened to cause that strange drop of portions of the text as it might reveal something more problematic. I will find some time to do that later and just apply this fix for now.